aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -731,6 +731,13 @@ static inline Tensor& bmm_out_or_baddbmm_(Tensor& self_or_result, const Tensor&
   checkOnCPU(batch1, c);
   checkOnCPU(batch2, c);
 
+  TORCH_CHECK(
```

Comment:
can you instead change accesses to .data_ptr<scalar_t>() in mkl/LinearAlgebra.cpp? Otherwise, baddbmm_mkl_template function is needlessly unsafe. You also need to either change `res.data_ptr<scalar_t>()` there (preferably), or check self_or_result dtype here.

Reply:
Fixed